src/tool.cpp

```diff
@@ -236,7 +248,23 @@ HitParams getHitParams(const ItemGroupList &armor_groups,
 				* armor / 100.0;
 	}
 
-	return {damage, 0};
+	for (const auto &groupcap : tp->groupcaps) {
+		const ToolGroupCap &cap = groupcap.second;
+
+		if (cap.uses != 0)
+			result_wear = 1.0 / cap.uses / pow(3.0, (double)(cap.maxlevel - 1.0));
+		else
+			result_wear = 0;
+	}
+
+	if (time_from_last_punch < tp->full_punch_interval) {
+		float f = time_from_last_punch / tp->full_punch_interval;
+		//infostream<<"f="<<f<<std::endl;
+		result_wear *= f;
+	}
+	u16 wear_i = 65535. * result_wear;
```

Comment:
I wonder if the decimal point is needed after '65535'?

Reply:
does the "u16" not mean this is an integer type?

Reply:
Multiplying two floats, and then converting the result to an integer results in better precision. Assuming `result_wear` is a float, it should be multiplied by `65535.0f` and then the result will be automatically converted to the type of the variable it's being assigned to.

Reply:
ahh, ok.